packages/sdk-v2/src/hooks/useMulticallAtChain.tsx

```diff
@@ -70,6 +71,21 @@ export async function multicall(
   return state;
 }
 
+export const getReadOnlyProvider = (chainId: number, readOnlyUrls: NodeUrls | undefined, pollingInterval: number) => {
+  if (readOnlyUrls && readOnlyUrls[chainId]) {
+    switch (true) {
+      case readOnlyUrls[chainId] instanceof BaseProvider:
+        return readOnlyUrls[chainId] as JsonRpcProvider;
+      case typeof readOnlyUrls[chainId] === "function":
+        return (readOnlyUrls[chainId] as any) as JsonRpcProvider;
+      case typeof readOnlyUrls[chainId] === "string":
+        const provider = new JsonRpcProvider(readOnlyUrls[chainId] as any);
+        provider.pollingInterval = pollingInterval;
+        return provider;
+    }
+  }
+}
+
```

Comment:
what is this for?
it seems like duplicate logic from useReadOnlyProvider. DRY

Reply:
This was part of trying to minimize re-renders, and tried to minimize hook usages
which one is preferred?

Reply:
bug with useEffect skipped in the useReadOnlyProvider is still unresolved